internal/fs/buffer/inmemory_write_buffer.go

```diff
@@ -0,0 +1,23 @@
+package buffer
+
+import (
+	"bytes"
+)
+
+type InMemoryWriteBuffer struct {
+	// buffer will hold the data to be written to GCS. Buffer holds the data for 2
+	// GCS write calls.
+	buffer *bytes.Buffer
+	// chunkSize is the size of data to be written in one GCS call.
+	chunkSize int
+}
+
+func (b *InMemoryWriteBuffer) Create(sizeInMB int) WriteBuffer {
```

Comment:
I'm unclear about this signature. I thought InMemoryWriteBuffer was supposed to implement the WriteBuffer interface, wasn't it? If yes, why is it returning a WriteBuffer from an instance of InMemoryWriteBuffer ?

Reply:
The function signatures should match for struct to implement an interface, that is why we were returning the interface instead of struct previously. 

However I have refactored to remove create method as suggested previously from the interface. PTAL again. Resolving all the comments related to this change.